stubs/Pagination.stub

```diff
@@ -28,8 +28,8 @@ abstract class AbstractPaginator implements \Illuminate\Contracts\Support\Htmlab
 /**
  * @template TValue
  *
- * @implements \ArrayAccess<mixed, TValue>
- * @implements \IteratorAggregate<mixed, TValue>
+ * @implements \ArrayAccess<array-key, TValue>
```

Comment:
Since this class implements `ArrayAccess` we should add the `offset*` methods here with the correct return/parameter types. This will let PHPStan know `User::paginate(10)[3]` returns `User` An example can be seen [here.](https://github.com/phpstan/phpstan-src/blob/bfcef682d335bf10d5c317aad53b0771276bfbd6/stubs/ArrayObject.stub)

Also for the other classes below + tests for this 👍🏽

Reply:
Done! ✅